workspace-d/source/workspaced/com/ccdb.d

```diff
@@ -21,61 +19,51 @@ class ClangCompilationDatabaseComponent : ComponentWrapper
 
 	protected void load()
 	{
-		logDebug("loading ccdb component");
-
-		if (config.get!bool("ccdb", "registerImportProvider", true))
-			importPathProvider = &imports;
-		if (config.get!bool("ccdb", "registerStringImportProvider", true))
-			stringImportPathProvider = &stringImports;
-		if (config.get!bool("ccdb", "registerImportFilesProvider", false))
-			importFilesProvider = &fileImports;
-		if (config.get!bool("ccdb", "registerProjectVersionsProvider", true))
-			projectVersionsProvider = &versions;
-		if (config.get!bool("ccdb", "registerDebugSpecificationsProvider", true))
-			debugSpecificationsProvider = &debugVersions;
-
-		try
-		{
-			if (config.get!string("ccdb", null))
-			{
-				const dbPath = config.get!string("ccdb", "dbPath");
-				if (!dbPath)
-				{
-					throw new Exception("ccdb.dbPath is not provided");
-				}
-				loadDb(dbPath);
-			}
-		}
-		catch (Exception e)
-		{
-			stderr.writeln("Clang-DB Error (ignored): ", e);
-		}
+		trace("loading ccdb component");
+
+		if (!refInstance)
+			throw new Exception("ccdb requires to be instanced");
+
+		importPathProvider = &imports;
+		stringImportPathProvider = &stringImports;
+		importFilesProvider = &fileImports;
+		projectVersionsProvider = &versions;
+		debugSpecificationsProvider = &debugVersions;
```

Comment:
why remove the ability to enable/disable the providers?

also this should call
```d
if (auto dbPath = config.get!string("ccdb", "dbPath", null))
	loadDb(dbPath);
```

to load dbPath immediately, if set.

Reply:
I had errors due to missing workspace instance. I'm only starting to get my head around the code base. 
I don't quite get yet how this configuration works, and when/where all the instances are created.
I will correct this as soon as I understand how these things work together.